### ISSUE TITLE
Build failure on rustc 1.20.0

### ISSUE THREAD

**Issue by jdanford** [ogham/exa#283]
I'm trying to install `exa` from source (with rustc 1.20.0 and macOS 10.12.6), and 3d4ddf8af6bd746b5d07b9ee62126fb404aa2c29 is the newest version I can build successfully; with the current version (265f93f7cd9b4560af64b606075a623b3fdf2105) I'm getting the following errors:
```rust
error[E0599]: no method named `has_anything_for` found for type `&fs::feature::git::GitCache` in the current scope
   --> src/output/details.rs:148:46
    |
148 |                 (Some(g), Some(d))  => if !g.has_anything_for(&d.path) { git = None },
    |                                              ^^^^^^^^^^^^^^^^

error[E0599]: no method named `has_anything_for` found for type `&fs::feature::git::GitCache` in the current scope
   --> src/output/details.rs:149:72
    |
149 |                 (Some(g), None)     => if !self.files.iter().any(|f| g.has_anything_for(&f.path)) { git = None },
    |                                                                        ^^^^^^^^^^^^^^^^

error[E0599]: no method named `has_anything_for` found for type `&fs::feature::git::GitCache` in the current scope
   --> src/output/grid_details.rs:172:42
    |
172 |             (Some(g), Some(d))  => if !g.has_anything_for(&d.path) { git = None },
    |                                          ^^^^^^^^^^^^^^^^

error[E0599]: no method named `has_anything_for` found for type `&fs::feature::git::GitCache` in the current scope
   --> src/output/grid_details.rs:173:68
    |
173 |             (Some(g), None)     => if !self.files.iter().any(|f| g.has_anything_for(&f.path)) { git = None },
    |                                                                    ^^^^^^^^^^^^^^^^

error[E0061]: this function takes 1 parameter but 2 parameters were supplied
   --> src/output/table.rs:353:28
    |
353 |               .map(|g| g.get(&file.path, file.is_directory()))
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 1 parameter
    |
   ::: src/fs/feature/mod.rs
    |
25  | /         pub fn get(&self, _index: &Path) -> Option<Git> {
26  | |             panic!("Tried to query a Git cache, but Git support is disabled")
27  | |         }
    | |_________- defined here

error[E0308]: mismatched types
   --> src/output/table.rs:352:9
    |
352 | /         self.git
353 | |             .map(|g| g.get(&file.path, file.is_directory()))
354 | |             .unwrap_or_default()
    | |________________________________^ expected struct `fs::fields::Git`, found enum `std::option::Option`
    |
    = note: expected type `fs::fields::Git`
               found type `std::option::Option<fs::feature::git::Git>`
```

**Comment by mqudsi** [comment]
Your title says rust 1.20.0 but your report says 1.12.0. Which is the typo?

**Comment by jdanford** [comment]
Whoops, just corrected it to 1.20.0!

**Comment by ogham** [comment]
This wasn't a problem with Rust 1.20.0 — it was a legitimate compile error! If I build with `--no-default-features` I got the exact same set of errors you see here, and a few days ago I changed some of the Git code but forgot to update the code that runs with Git support disabled.

It happened because I don't test building with `--no-default-features` as often as I should, so the "real" fix was to have Travis test this every time. Now I'll get alerted the next time I break it.
